src/custom-blocks/featured-item/edit.js

```diff
@@ -0,0 +1,380 @@
+import {
+	PanelBody,
+	SelectControl,
+	TextControl,
+	ToggleControl,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import {
+	InnerBlocks,
+	MediaUpload,
+	InspectorControls,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+const { Fragment } = wp.element;
+const d = new Date();
+const allowedMediaTypes = ['image'];
+const templatefeaturedItemBlock = [
+	[ 'core/heading', { placeholder: 'Add featured item section title' } ]
+];
+const noItemSelectedText = "No item selected"; // Text in editor to shew that no item will be displayed
+
+export default function featuredItemEdit({ attributes, setAttributes} ) {
+
+	const {
+		featuredImage,
+		featuredItemType,
+		featuredItemID,
+		featuredItemHasDate,
+		featuredItemHasBar,
+		featuredLinkText,
+		featuredCustomImage,
+		className,
+	} = attributes;
+
+	const {
+		allPostTypes,
+	} = useSelect(
+		( select ) => {
+
+			const { getPostTypes } = select(
+				coreStore
+			);
+			const allPostTypeThingies = getPostTypes(
```

Comment:
Thingies?

Reply:
Changed - isn't naming variables the hardest thing in the world!??!?